ipython_notebook_config.txt

```diff
@@ -42,12 +42,14 @@
 #
 #In jupyter_notebook_config.py, we insert the following Python code:
 
+from __future__ import print_function
 import sys
 import json
 import os
 
-reload(sys)
-sys.setdefaultencoding('utf8')
+if (sys.version_info < (3,0)):
```

Comment:
I would replace this black magic with a simple (sys.version_info[0] < 3).  Comparing tuples of unequal lengths are against the will of gods... :)  (It would be even better to use sys.version.major < 3, but I think that in Python 3.0 the sys.version_info tuple has no named attributes.)